### PR TITLE
Support GitHub Codespaces

### PR DESCRIPTION
**Pull request by AdColvin**
### Context
We wanted to give new developers a way to get a development environment running, if they're having issues on their own machines.  

### Changes proposed in this pull request
This allows devs to create a codespace through the GitHub UI, which will give them a VSCode window either in the browser or through the VSCode app. They can then start a development server with `bin/dev` 

I've added a separate Dockerfile and docker-compose file in the .devcontainer directory, so as not to mess anything currently being used in production. 

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+version: '3'
+services:
+  web:
+    build:
+      context: ..
+      dockerfile: .devcontainer/Dockerfile
+    command: sleep infinity
+    ports:
+      - "3000:3000"
```
Comment (defong) [inline .devcontainer/docker-compose.yml:9]: Should it be port 5000?
https://github.com/DFE-Digital/register-trainee-teachers/blob/c057344eefa488c3b00eca84bed19be3487da2b1/config/settings.yml#L1

**Review by tomtrentham — COMMENTED** [review]
Good to see this working. Probably things we can build on here, depending on what our main use case is. I think it would be nice to have some minimal docs on this setup.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ class ApplicationController < ActionController::Base
   before_action :check_organisation_context_is_set
   before_action :set_sentry_organisation_context, unless: -> { Rails.env.local? }
   after_action :save_origin_path
+  if Rails.env.development? then skip_before_action :verify_authenticity_token end
```
Comment (defong) [inline app/controllers/application_controller.rb:14]: Hmmm are we sure?

**Review by defong — APPROVED** [review]

File: app/controllers/concerns/codespaceable.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  GITHUB_APP_DOMAIN = "app.github.dev".freeze
+
+  included do
+    skip_before_action :verify_authenticity_token, if: :running_in_codespace?
+  end
+
+  private
+
+    def running_in_codespace?
+      Rails.env.development? && request.host.contains?(GITHUB_APP_DOMAIN)
```
Comment (tomtrentham) [inline app/controllers/concerns/codespaceable.rb:15]: Could we match the end of the host to make this a little bit safer?
```suggestion
      Rails.env.development? && request.host.ends_with?(GITHUB_APP_DOMAIN)
```

**Review by tomtrentham — APPROVED** [review]

**Pull request merged**